### ISSUE TITLE
🐛 Bug Report: "Invalid `success` param: URL host must be one of ...

### ISSUE THREAD

**Issue by erdkse** [appwrite/appwrite#7255]
### 👟 Reproduction steps

I am a self-hosted user. I installed appwrite with domain 'A' and created an application and added another domain 'B' as custom domain to project to support OAuth login.

When I try to login via Google I receive this error. 

<img width="616" alt="Screenshot 2023-12-07 at 11 27 54" src="https://github.com/appwrite/appwrite/assets/2957624/01f334dd-59c3-464d-9098-6746cb16d01b">

I configured SDK platform hostname correctly and I am sending right redirect urls on success and fail cases when logging in.

### 👍 Expected behavior

It should succesfully login.

### 👎 Actual Behavior

It throws an error for misconfiguration.

### 🎲 Appwrite version

Version 1.4.x

### 💻 Operating system

Linux

### 🧱 Your Environment

I use Appwrite v1.4.9 on docker-compose. There is no customization.

I am using Web SDK v13.0.1

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@erdkse, thanks for creating this issue! 🙏 Have you created a web platform with the same hostname as the success url you're trying?

**Comment by erdkse** [comment]
Yes @stnguyen90, email session work well on my actual domain, but there is another issue with the hostname, If I add only the domain I am not able to create an email session too, I have to add hostname with wildcard as `*.B`.

To summarize again;

1. I installed appwrite instance to `appwrite.A` domain
2. I created a project and added a custom domain `appwrite.B`.
3. I configured my Web platform hostname as `*.B`.
4. I am interacting to appwrite instance pointing `appwrite.B` by Web SDK.
4. My web application is getting served directly from domain `B`.

And the error message says that. The success URL must be one of `localhost`, `appwrite.B`, `*.B`.

If I change Web platform hostname to `B`.  Error message changes as `localhost`, `appwrite.B`, `B`. But this time creating email session throws CORS error.

![Screenshot 2023-12-08 at 19 47 53](https://github.com/appwrite/appwrite/assets/2957624/da0f8e2a-288d-4c97-b312-f3950f56d181)

It looks like my Request Headers' origin and the rest are correct when I send request from domain `B` but I receive this Response Headers.

![Screenshot 2023-12-08 at 19 51 35](https://github.com/appwrite/appwrite/assets/2957624/1073d0de-f6c2-4876-a154-e0d1374cd8f9)


Note: There is no issue when creating OAuth or Email session on localhost.

**Comment by optionedge** [comment]
I also facing exact same issue. My hosting and usage environment is similar to above. I am also getting the following message:

```Error 400
Invalid `success` param: URL host must be one of: localhost, auth.mydomain.in, *

Type

general_argument_invalid
```

**Comment by stnguyen90** [comment]
@erdkse, thanks for the details. It should work if you put an exact match for the domain your app is running on and the web platform. For example, if my app is running on app.example.com, I would put app.example.com as the web platform. In addition, my success url for the create OAuth2 session API call would be something like `https://app.example.com`.

Btw, there is a problem in which wildcards are not honored when validating URLs in params like the success URL. That issue is here: https://github.com/appwrite/appwrite/issues/6322#issuecomment-1850534416.

**Comment by stnguyen90** [comment]
@optionedge, are you still seeing the error? FYI, I'll need to close this due to inactivity soon.

**Comment by erdkse** [comment]
Sorry @stnguyen90, I have some unexpected things to do so I am not able to share logs for the next 2-3 days. Yes, I am still having the issue but I was able to solve the problem by adding `www` in front of my hostname.

I will share the logs as soon as I find the chance.

**Comment by optionedge** [comment]
@stnguyen90  If I put exact domain,  in web platform and put the same in success url for oauth then it works, however then email login stopped working because of cors issue:

This is my current scenario:

1. Email Registration and Login
I have authentication hosted on auth.website.in (appwrite), my landing/login page is on website.in. When I put exact domain name in web platform I got cors issue:

`Access to XMLHttpRequest at 'https://auth.mywebsite.in/v1/account/sessions/email' from origin 'https://mywebsite.in' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header has a value 'https://localhost' that is not equal to the supplied origin.`

As soon as I change the host on web platform to '*' email login works properly, then it breaks the oAuth.

3. oAuth Login
As explained above, oauth works when put exact domain in web platform host, but it breaks the email login.  I am distributing software that can be installed at clients end as web application. When oauth is successful, I am redirecting to localhost (localhost in success url). Appwrites check for exact domain name (not wildcard, stopping me to use oAuth). 

I have currently disabled the oauth and using only email auth.   Not sure how to get around the above issue.

**Comment by stnguyen90** [comment]
@erdkse, did you add an additional web platform for your site. Seeing as you have 2 websites accessing your Appwrite project, you should have 2 web platforms with the matching hostname.

**Comment by erdkse** [comment]
hey @stnguyen90, no I have only 1 platform to reach the project, the other hostname is responsible for reaching to appwrite in general or to the other projects inside appwrite. But I solved my problem only by adding "www" in front of my hostname. Now I am able to handle both email and OAuth authentication without any issue.
